### DISCUSSION TITLE
Database requests in goroutines

### DISCUSSION THREAD

**Discussion by maxbry** [gobuffalo/buffalo#2082] [Q&A]
Hello,

in my requests, some things need to be stored in the database for logging reasons. I would like to do this in goroutines because this can happen concurrently:

    doStuff()
    go storeLogs(tx, data)
    doOtherStuff()

I noticed that somethings the goroutine wanted to access the database after the original request had been finished. The tx seems to could not have been usable anymore. What is the best practice in this case? Should I create a new database connection inside the goroutine?

**Comment by paganotoni** [comment]
Hey @maxbry. 
As the request may have ended, the transaction on the request would be committed or rollbacked by the time the execution of the goroutine starts. 

One thing you could do is using models.DB to start a new transaction.
